bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusProvider.java

```diff
@@ -0,0 +1,475 @@
+     *
+     * @param pConnect true = make a new connection , false = close existing connection
+     * @return boolean true = connection succeeded, false = connection failed
+     */
+    private synchronized boolean connSocketConnect() {
+        logger.debug(" connSocketConnect '{}' - attempting connection to host '{}', port '{}'",
+                tivoConfigData.getCfgIdentifier(), tivoConfigData.getHost(), tivoConfigData.getTcpPort());
+
+        if (connIsConnected()) {
+            logger.debug(" connSocketConnect '{}' - already connected to host '{}', port '{}'",
+                    tivoConfigData.getCfgIdentifier(), tivoConfigData.getHost(), tivoConfigData.getTcpPort());
+            return true;
+        } else {
+            // something is wrong, so force a disconnect/clean up so we can try again
+            connTivoDisconnect(true);
+        }
+
+        try {
+            Socket tivoSocket = new Socket(tivoConfigData.getHost(), tivoConfigData.getTcpPort());
+            tivoSocket.setKeepAlive(true);
+            tivoSocket.setSoTimeout(CONFIG_SOCKET_TIMEOUT_MS);
+            tivoSocket.setReuseAddress(true);
+
+            if (tivoSocket.isConnected() && !tivoSocket.isClosed()) {
+                if (streamWriter == null) {
+                    streamWriter = new PrintStream(tivoSocket.getOutputStream(), false);
+                }
+                if (this.streamReader == null) {
+                    StreamReader streamReader = new StreamReader(tivoSocket.getInputStream());
+                    streamReader.start();
+                    this.streamReader = streamReader;
+                }
+                this.tivoSocket = tivoSocket;
+            } else {
+                logger.debug(" connSocketConnect '{}' - socket creation failed to host '{}', port '{}'",
+                        tivoConfigData.getCfgIdentifier(), tivoConfigData.getHost(), tivoConfigData.getTcpPort());
+                return false;
+            }
+
+            return true;
+
+        } catch (UnknownHostException e) {
+            logger.debug(" TiVo '{}' - while connecting, unexpected host error: '{}'",
+                    tivoConfigData.getCfgIdentifier(), e.getMessage());
+        } catch (IOException e) {
+            if (tivoStatusData.getConnectionStatus() != ConnectionStatus.OFFLINE) {
+                logger.debug(" TiVo '{}' - I/O exception while connecting: '{}'", tivoConfigData.getCfgIdentifier(),
+                        e.getMessage());
+            }
+        }
+        return false;
+    }
+
+    /**
+     * {@link doNappTime} sleeps for the period specified by the getCmdWaitInterval parameter. Primarily used to allow
+     * the TiVo time to process responses after a command is issued.
+     */
+    public void doNappTime() {
+        try {
+            logger.debug(" doNappTime '{}' - I feel like napping for '{}' milliseconds",
+                    tivoConfigData.getCfgIdentifier(), tivoConfigData.getCmdWaitInterval());
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging